lib/utils/class-alerts.php

```diff
@@ -0,0 +1,270 @@
+		}
+
+		return $channel_or_user;
+	}
+
+	/**
+	 * Validate alert message
+	 *
+	 * Ensure the message is a valid string
+	 *
+	 * @param $message mixed
+	 *
+	 * @return string Validated and trimmed message
+	 */
+	protected function validate_message( $message ) {
+		if ( is_array( $message ) || is_object( $message ) ) {
+			throw new Exception( "Invalid \$message: Alerts\:\:chat( " . print_r( $message, true ) . " );" );
+		}
+
+		$message = trim( $message );
+
+		if ( ! $message ) {
+			throw new Exception( "Invalid \$message: Alerts\:\:chat( '$message' );" );
+		}
+
+		return $message;
+	}
+
+	/**
+	 * Validate Opsgenie details array
+	 *
+	 * @param $details array
+	 *
+	 * @return array
+	 */
+	protected function validate_details( $details) {
+		$required_keys = [ 'alias', 'description', 'entity', 'priority', 'source' ];
+
+		if ( ! is_array( $details ) ) {
+			throw new Exception( "Invalid \$details: Alerts\:\:opsgenie( " . print_r( $details, true ) ." );" );
+		}
+
+		foreach( $details as $key => $value ) {
+			if ( ! array_key_exists( $key, $required_keys ) ) {
+				throw new Exception( "Invalid \$details: Alerts\:\:opsgenie( " . print_r( $details, true ) ." );" );
+			}
+
+			if ( ! $value ) {
+				throw new Exception( "Invalid \$details: Alerts\:\:opsgenie( " . print_r( $details, true ) ." );" );
+			}
+		}
+
+		return $details;
+	}
+
+	/**
+	 * Get an instance of this Alerts class
+	 *
+	 * @return Alerts
+	 */
+	public static function get_instance() {
+		if ( null === self::$instance ) {
+			self::$instance = new Alerts();
+		}
+
+		return self::$instance;
+	}
+
+	/**
+	 * Send a message to IRC
+	 *
+	 * $level can be an int of one of the following
+	 * NONE = 0
+	 * WARNING = 1
+	 * ALERT = 2
+	 * CRITICAL = 3
+	 * RECOVERY = 4
+	 * INFORMATION = 5
+	 * SCALE = 6
+	 *
+	 * Example Usage
+	 *
+	 * Alerts::chat( '@testuser', 'test message' );			// send testuser a pm on IRC from "a8c"
+	 * Alerts::chat( '@testuser', 'test message', 3 );	// send testuser a pm on IRC with level 'critical'
+	 * Alerts::chat( 'testing', 'test message' );				// have "a8c" join #testing and say something
+	 * Alerts::chat( 'testing', 'test message', 4 );		// have "a8c-test" join #testing and say something with level 'recovery'
+	 *
+	 * @param $target (string) Channel or Username.  Usernames prefixed with an @, channel optionally prefixed by #.
+	 * @param $message (string) Message
+	 * @param $level (int) Level The severity level of the message
+	 * @param $kind string Cache slug
+	 * @param $interval integer Interval in seconds between two messages sent from one DC
+	 *
+	 * @return bool	True if successful. Else, will return false
+	 */
+	public static function chat( $channel_or_user, $message, $level = 0, $kind = '', $interval = 0 ) {
+		try {
+			$alerts = self::get_instance();
+
+			if ( $kind && $interval ) {
+				if ( ! $alerts->add_cache( $kind, $interval ) ) {
+					return false;
```

Comment:
We should log this case

Reply:
Added in 0625bb47df5c87019355abd8cd35c3769f6461f5